src/main/java/com/shaft/support/DBActions.java

```diff
@@ -0,0 +1,155 @@
+
+	public DBActions(String dbType, String dbServerIP, String dbPort, String dbName, String username, String password) {
+		this.dbType = dbType;
+		this.dbServerIP = dbServerIP;
+		this.dbPort = dbPort;
+		this.dbName = dbName;
+		this.username = username;
+		this.password = password;
+	}
+
+	private void passAction(String actionName, String testData, String log) {
+		String message = "Successfully performed action [" + actionName + "].";
+		if (testData != null) {
+			message = message + " With the following test data [" + testData + "].";
+		}
+		ReportManager.log(message);
+		if (log != null) {
+			ReportManager.attachAsStep("Passed", "DB Connection Log", log);
+		}
+	}
+
+	private void passAction(String actionName, String testData) {
+		passAction(actionName, testData, null);
+	}
+
+	private void failAction(String actionName, String testData, String log) {
+		String message = "Failed to perform action [" + actionName + "].";
+		if (testData != null) {
+			message = message + " With the following test data [" + testData + "].";
+		}
+		ReportManager.log(message);
+		if (log != null) {
+			ReportManager.attachAsStep("Failed", "DB Connection Log", log);
+		}
+		Assert.fail(message);
+	}
+
+	private void failAction(String actionName, String testData) {
+		failAction(actionName, testData, null);
+	}
+
+	/**
+	 * Setup connection to database
+	 * 
+	 * @throws ClassNotFoundException
+	 * @throws SQLException
+	 */
+	public void createConnection() {
+		String testData = dbServerIP + ":" + dbPort + "/" + dbName;
+
+		try {
+			switch (dbType) {
+			// create connection to mysql DB
+			case ("mysql"):
+				Class.forName("com.mysql.jdbc.Driver");
+				connection = DriverManager.getConnection("jdbc:mysql://" + dbServerIP + ":" + dbPort + "/" + dbName,
+						username, password);
+				break;
+			// create connection to SQL Server DB
+			case ("sqlserver"):
+				Class.forName("com.microsoft.sqlserver.jdbc.SQLServerDriver");
+				connection = DriverManager.getConnection(
+						"jdbc:sqlserver://" + dbServerIP + ":" + dbPort + ";databaseName=" + dbName, username,
+						password);
+				break;
+			// create connection to SQL Server DB
+			case ("postgresql"):
+				Class.forName("org.postgresql.Driver");
+				connection = DriverManager.getConnection("jdbc:postgresql://" + dbServerIP + ":" + dbPort + "/" + dbName,
+						username, password);
+				break;
+			// create connection to oracle DB
+			case ("oracle"):
+				break;
+			}
+
+			statement = connection.createStatement();
+			passAction("create DB connection", testData);
+		} catch (SQLException e) {
+			ReportManager.log(e);
+			failAction("create DB connection", testData);
+		} catch (ClassNotFoundException e) {
+			ReportManager.log(e);
+			failAction("create DB connection", testData);
+		}
+	}
+
+	/**
+	 * execute DB query and return the result as ResultSet object
+	 * 
+	 * @param dbQuery
+	 * @return ResultSet
+	 * @throws SQLException
+	 */
+	public ResultSet executeSelectQuery(String dbQuery) {
```

Comment:
it is a must to return a regular String to be consistent with the rest of the methods that do not require further processing of the returned data via special methods.

Reply:
This will be handled in POM class because the query result is not returned as string, you have to make some processing of ResultSet object to get your data which will be implemented on another layer as explained in the meeting

Reply:
Please plan a meeting and let's discuss any missing points.
For this point, no extra processing should be required after using any SHAFT_Engine method. In the APIs we return the rest object, as SHAFT itself is used to process it and extract json or xml values from it. So if further processing is needed, helping methods that extract the values should be implemented in SHAFT and then used in the ObjectModel class. If not then SHAFT should return a string. Eitherway SHAFT should make an attachment with the returned result from the action. The same happens in the GUI/CLI/API sections.

Reply:
will reopen on the latest commit.